### PR TITLE
Have consistent naming for Google Cloud services

### LINKED ISSUE #1485: Have consistent naming for Google Cloud services
The cloud semantic conventions specify a number of Google Cloud services  https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/resource/cloud.yaml#L74 but they are not consistently named.

For example, 'gcp_gke' uses a short name for the service whereas 'gcp_compute_engine' and 'gcp_cloud_functions' uses the long name. Either use short names everywhere or switch to the long names for all.

cc @jsuereth

### PR DESCRIPTION
**Pull request by rakyll**
`gcp_gke` is the only value that contains a shorten version of the
cloud service in cloud semantic conventions.
Use the long name to have consistency with others.

Also update the briefs the use the fully qualified names
of Google Cloud services.

Fixes #1485.

### CODE REVIEW

File: semantic_conventions/resource/cloud.yaml [hunk 1]
```diff
@@ -78,19 +78,19 @@ groups:
...
+              brief: Google Compute Engine
             - id: GCP_CloudRun
               value: 'gcp_cloud_run'
-              brief: GCP Cloud Run
-            - id: GCP_GKE
-              value: 'gcp_gke'
+              brief: Google Cloud Run
+            - id: GCP_KubernetesEngine
+              value: 'gcp_kubernetes_engine'
               brief: Google Kubernetes Engine
```
Comment (Oberon00) [inline semantic_conventions/resource/cloud.yaml:87]:
```suggestion
              brief: Google Cloud Kubernetes Engine
```

File: semantic_conventions/resource/cloud.yaml [hunk 2]
```diff
@@ -78,19 +78,19 @@ groups:
               brief: Azure App Service
             - id: GCP_ComputeEngine
               value: 'gcp_compute_engine'
-              brief: GCP Compute Engine
+              brief: Google Compute Engine
```
Comment (Oberon00) [inline semantic_conventions/resource/cloud.yaml:81]:
```suggestion
              brief: Google Cloud Compute Engine
```

**Review by Oberon00 — APPROVED** [review]

File: semantic_conventions/resource/cloud.yaml (hunk 2 above)
Comment (arminru) [reply semantic_conventions/resource/cloud.yaml:81]: Their brand name is actually Google Kubernetes/Compute Engine (GKE and GCE) without "Cloud" so it should be fine.

Comment (arminru) [inline semantic_conventions/resource/cloud.yaml:81]:
```suggestion
              brief: Google Compute Engine (GCE)
```

File: semantic_conventions/resource/cloud.yaml (hunk 1 above)
Comment (arminru) [inline semantic_conventions/resource/cloud.yaml:87]:
```suggestion
              brief: Google Kubernetes Engine (GKE)
```

Comment (arminru) [reply semantic_conventions/resource/cloud.yaml:87]: same here

File: semantic_conventions/resource/cloud.yaml [hunk 3]
```diff
@@ -78,19 +78,19 @@ groups:
...
+              value: 'gcp_kubernetes_engine'
               brief: Google Kubernetes Engine
             - id: GCP_CloudFunctions
               value: 'gcp_cloud_functions'
-              brief: GCP Cloud Functions
+              brief: Google Cloud Functions
             - id: GCP_AppEngine
               value: 'gcp_app_engine'
-              brief: GCP App Engine
+              brief: Google Cloud App Engine
```
Comment (arminru) [inline semantic_conventions/resource/cloud.yaml:93]:
```suggestion
              brief: Google App Engine (GAE)
```

**Review by arminru — APPROVED** [review]

File: semantic_conventions/resource/cloud.yaml (hunk 2 above)
Comment (jsuereth) [reply semantic_conventions/resource/cloud.yaml:81]: +1

File: semantic_conventions/resource/cloud.yaml (hunk 1 above)
Comment (jsuereth) [reply semantic_conventions/resource/cloud.yaml:87]: +1 to @arminru comment, see: https://cloud.google.com/kubernetes-engine for the branded product name

Comment (jsuereth) [reply semantic_conventions/resource/cloud.yaml:87]: I think we should be consistent with acronym usage across this whole file.  EKS, e.g. is branded as [Amazon Elastic Kubernetes Engine](https://aws.amazon.com/eks/?whats-new-cards.sort-by=item.additionalFields.postDateTime&whats-new-cards.sort-order=desc&eks-blogs.sort-by=item.additionalFields.createdDate&eks-blogs.sort-order=desc).  If we're going to call out an acronym here, is there a reason we wouldn't use it in the `value` field to be consistent across cloud offerings?

Just want to understand what judgement call / criteria we're using for naming here so we're consistent across all offerings.

**Review by SergeyKanzhelev — APPROVED** [review]

**Review by reyang — APPROVED** [review]
LGTM.

File: semantic_conventions/resource/cloud.yaml (hunk 2 above)
Comment (rakyll) [reply semantic_conventions/resource/cloud.yaml]: Done for all.

Comment (rakyll) [reply semantic_conventions/resource/cloud.yaml]: Done.

**Review by rakyll — COMMENTED** [review]
PTAL

**Comment by carlosalberto** [comment]
@jsuereth does it look good to go?

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**